build.zig

```diff
@@ -106,6 +105,131 @@ fn generateWasmBindings(b: *Build, params: GenerateWasmBindingsParams) *Build.St
     return copy_outputs_to_src;
 }
 
+const OrcaAppBuildParams = struct {
+    name: []const u8,
+    sources: []const []const u8,
+    install: []const u8, // Orca app will be bundled to zig-out/{install}/{name}
+
+    icon_path: ?[]const u8 = null,
+    resource_path: ?[]const u8 = null,
+    shaders: ?[]const []const u8 = null,
+    create_run_step: bool = false, // if false or the target is not the host platform, no run step will be generated
+};
+
+const OrcaAppBuildSteps = struct {
+    build_or_bundle: *Step,
+    run: ?*Step.Run,
+};
+
+fn buildOrcaApp(
+    b: *Build,
+    target: Build.ResolvedTarget,
+    wasm_sdk_lib: *Step.Compile,
+    wasm_libc_lib: *Step.Compile,
+    gen_header_exe: *Step.Compile,
+    orca_install: *Step,
+    params: OrcaAppBuildParams,
+) OrcaAppBuildSteps {
+    var wasm_target_query: std.Target.Query = .{
+        .cpu_arch = std.Target.Cpu.Arch.wasm32,
+        .os_tag = std.Target.Os.Tag.freestanding,
+    };
+    wasm_target_query.cpu_features_add.addFeature(@intFromEnum(std.Target.wasm.Feature.bulk_memory));
+    wasm_target_query.cpu_features_add.addFeature(@intFromEnum(std.Target.wasm.Feature.nontrapping_fptoint));
+
+    const wasm_target: Build.ResolvedTarget = b.resolveTargetQuery(wasm_target_query);
+    const wasm_optimize: std.builtin.OptimizeMode = .ReleaseSmall;
+
+    const wasm_module = b.addExecutable(.{
+        .name = "module",
+        .linkage = .static,
+        .root_module = b.createModule(.{
+            .target = wasm_target,
+            .optimize = wasm_optimize,
+            .single_threaded = true,
+            .link_libc = false,
+            .strip = true, // setting to true removes all embedded DWARF info
```

Comment:
We might want to switch this to `false`, especially for the debugger branch.

Reply:
For the samples it feels like they should just always have debug symbols because the point of them is to demo what's possible with Orca, which includes showing off the debugger. I'll just make sure the debug symbols aren't stripped unless you feel we should still provide a user-specified option to strip them.

Reply:
This has been set to `false` as discussed